arwen/contexts/output.go

```diff
@@ -201,6 +201,14 @@ func (context *outputContext) ClearReturnData() {
 	context.outputState.ReturnData = make([][]byte, 0)
 }
 
+func (context *outputContext) RemoveReturnData(index uint32) {
+	returnData := context.outputState.ReturnData
+	if index >= uint32(len(returnData)) {
+		return
+	}
+	context.outputState.ReturnData = append(returnData[:index], returnData[index+1:]...)
```

Comment:
Would it make sense to call `ClearReturnData` if `index` is 0 and `len(returnData) == 1`? What happens here if we remove the last element of `returnData`? Hopefully, `append` will return an empty slice, and won't return `nil`.

Reply:
Your hopes are true :) A test was just added.